### PR TITLE
added NOAAs UPP package

### PR DESCRIPTION
**Pull request by edwardhartnett**
added NOAAs UPP package

### CODE REVIEW

File: var/spack/repos/builtin/packages/upp/package.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    package designed to generate useful products from raw model
+    output.
+    """
+
+    homepage = "https://github.com/NOAA-EMC/UPP"
+    git = "https://github.com/NOAA-EMC/UPP.git"
+
+    maintainers = ['kgerheiser', 'edwardhartnett', 'Hang-Lei-NOAA']
+
+    version('10.0.9', branch='upp_v10.0.9', submodules=True)
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/upp/package.py]: The general practice for numbered branches is to use the `tag` or, less often, the commit.  The former would look like:

```suggestion
    version('10.0.9', tag='upp_v10.0.9', submodules=True)
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]

**Comment by t-brown** [comment]
Quick question, has EMC stored out the naming issues? As in there was [EMC_post](https://github.com/NOAA-EMC/EMC_post), with documentation referring to itself as UPP and the final binary built was `ncep_post`.

If we want to have UPP then I am all for it and will deprecate the [ncep_post](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/ncep-post/package.py) package.

**Comment by edwardhartnett** [comment]
Sorry, I was not aware of the ncep_post package.

EMC_Post officially changed names to "UPP". Documentation should use UPP. It would be best if the spack name were "upp" as well.

If that can be more easily accommodated by a change to the existing ncep_post package, let me know.

**Comment by t-brown** [comment]
I'm happy pulling [ncep-post](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/ncep-post/package.py), as this UPP package has more `variant` options. I also see you've cleaned up the installed binary installed name to `upp.x` (I still really hate the use of that suffix) and pulled in a common CMake modules (so the package doesn't need to be patched).

**Comment by edwardhartnett** [comment]
Looks like this CI has gotten stuck - how to unstick it?

**Comment by tldahlgren** [comment]
@spackbot run pipeline

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**